src/pybind/mgr/cephadm/module.py

```diff
@@ -2473,6 +2473,11 @@ def _apply_service_spec(self, spec: ServiceSpec) -> str:
             raise OrchestratorError((f'The maximum count_per_host allowed is {max_count}.'
                                      + ' This limit can be adjusted by changing the mgr/cephadm/max_count_per_host config option'))
 
+        # prometheus is special case, we need to check prometheus module option for port
+        if spec.service_type == 'prometheus':
+            monitoring_spec = cast(MonitoringSpec, spec)
+            monitoring_spec.port = cast(int, self._ceph_get_module_option('prometheus', 'server_port'))
```

Comment:
Does this still work if we use `self.get_module_option_ex('prometheus', 'server_port', 9095)` instead? If so, I think it would be preferable. Generally nicer to use the api in `mgr_module.py` then directly calling the underlying functions.

Reply:
I updated the code following your recommendation and tested the change. It works :+1: